src/us/kbase/assemblyhomology/load/Loader.java

```diff
@@ -77,26 +82,59 @@ public void load(
 			final LoadID loadID,
 			final MinHashImplementation minhashImpl,
 			final MinHashDBLocation sketchDBlocation,
+			final Set<MinHashDistanceFilterFactory> filters,
 			final Restreamable namespaceYAML,
 			final Restreamable sequenceMetaJSONLines)
 			throws MinHashException, IOException, LoadInputParseException,
 				AssemblyHomologyStorageException {
+		// reaaaaally need to think about a builder here
 		checkNotNull(loadID, "loadID");
 		checkNotNull(minhashImpl, "minhashImpl");
 		checkNotNull(sketchDBlocation, "sketchDBlocation");
+		checkNoNullsInCollection(filters, "filters");
 		checkNotNull(namespaceYAML, "namespaceYAML");
 		checkNotNull(sequenceMetaJSONLines, "sequenceMetaJSONLines");
 		final NamespaceLoadInfo nsinfo = loadNameSpaceInfo(namespaceYAML);
 		final MinHashSketchDatabase sketchDB = minhashImpl.getDatabase(
 				new MinHashSketchDBName(nsinfo.getId().getName()), sketchDBlocation);
 		final Set<String> seqmetaIDs = loadSequenceMetaIDs(sequenceMetaJSONLines);
 		final Set<String> skdbIDs = new HashSet<>(minhashImpl.getSketchIDs(sketchDB));
+		validateSequenceIDs(filters, nsinfo, skdbIDs);
 		checkEqual(seqmetaIDs, sequenceMetaJSONLines, skdbIDs, sketchDBlocation);
 		loadSeqs(nsinfo.getId(), loadID, sequenceMetaJSONLines);
 		//TODO CODE set same load time for all seqs as option
 		storage.createOrReplaceNamespace(nsinfo.toNamespace(sketchDB, loadID, clock.instant()));
 	}
 
+	private void validateSequenceIDs(
```

Comment:
This invalidates sequence ids, rather than validates them.

Reply:
I don't think that's the case - if the method returns normally, it means the IDs are valid. It's only under exceptional conditions that IDs are invalidated. Happy to make changes, just not seeing it in this case.